modules/unsupported/geojson-store/src/main/java/org/geotools/data/geojson/store/GeoJSONDataStore.java

```diff
@@ -61,8 +62,8 @@ public GeoJSONDataStore(File f) {
     }
 
     GeoJSONReader read() throws IOException {
-        GeoJSONReader reader = new GeoJSONReader(getUrl());
-
+        GeoJSONReader reader =
+                new GeoJSONReader(getUrl(), FilenameUtils.getBaseName(url.getPath()));
```

Comment:
Why is this necessary?

Reply:
To make it prefix the ids with the file name, and retain the same behavior.

Reply:
but that is what the `new GeoJSONReader(getUrl());` call did any way - so I don't see any need for this

Reply:
Ah, did not notice, sorry.

Reply:
Yeah, I added the second argument as a means to seperate the featurebuilder name from the prefix.  If those should always be the same I can remove the extra constructor argument.

Reply:
The idea of a 2nd argument is fine, it just isn't needed here. as if you don't supply the 2nd argument the other constructor will make a baseName call.

Reply:
Ahh, gotcha.  It is now needed because without the second argument, if there is an existing id field, no prefix is added. In cases where there is no existing id field, it behaves as before and defaults to adding the derived prefix.

Reply:
As I said, I understand why you have provided a constructor with 2 arguments - my question is why in this exact line is it necessary to pass in the basename as the 2nd argument when the single argument constructor will do exactly the same thing?

Reply:
Now changed so the default setting mean that this is no longer needed, as requested.